### PR TITLE
[npm] Update validator to v8.1

### PR DESCRIPTION
**Pull request by jeremywiebe**
Updates the `validator` npm package to v8.1 (which adds `isPostalCode` validator function)

 **JIRA**: (link to JIRA ticket)
 **Linked PRs**: (links to corresponding PRs, optional)

## Changes
- (change1)

## Todos
- [ ] (if applicable) analytics events instrumented to measure impact of change
- [ ] Add a high-level description of your changes to the CHANGELOG.md, including a link to the PR with your changes

## How to test-drive this PR
- (necessary config changes)
- Run `npm start`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- (how to access the new / changed functionality -- fixtures, URLs)

### CODE REVIEW

File: web/app/containers/checkout-shipping/partials/checkout-shipping-form.jsx [hunk 1]
```diff
@@ -53,6 +54,10 @@ const validate = (values, props) => {
         errors.username = REQUIRED_TEXT
     }
 
+    if (values.countryId && values.postcode && !isPostalCode(values.postcode, values.countryId.toUpperCase())) {
```
Comment (kevinxh) [inline web/app/containers/checkout-shipping/partials/checkout-shipping-form.jsx]: There are `countryId`s that are unsupported according to Validator's doc. (see image below)

In shipping address form, if you select a third world country such as `São Tomé and Príncipe`, type some random region, then type postal code, Validator will throw error and the postal code field is never filled.

screen shot 2017-09-07 at 4 55 41 pm

screen shot 2017-09-07 at 4 48 11 pm

Comment (jeremywiebe) [reply web/app/containers/checkout-shipping/partials/checkout-shipping-form.jsx]: Good catch!  I'll fix that up to just not do validation for those for now.

File: web/app/utils/validation.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ */
+export const validatePostalCode = (postalCode, countryId) => {
+    if (!countryId) {
+        throw new Error('`countryId` must be supplied to this function')
+    }
+
+    try {
+        return isPostalCode(postalCode, countryId)
+    } catch (e) {
+        console.error(
```
Comment (kevinxh) [inline web/app/utils/validation.js]: 👍

File: web/app/modals/cart-estimate-shipping/container.jsx [hunk 3]
```diff
@@ -6,6 +6,7 @@ import React from 'react'
 import {connect} from 'react-redux'
 import * as ReduxForm from 'redux-form'
 import {createPropsSelector} from 'reselect-immutable-helpers'
+
```
Comment (kevinxh) [inline web/app/modals/cart-estimate-shipping/container.jsx]: 💯

File: web/app/containers/account-address/partials/address-form.jsx [hunk 4]
```diff
@@ -6,11 +6,14 @@ import React from 'react'
 import {connect} from 'react-redux'
 import {createPropsSelector} from 'reselect-immutable-helpers'
 import * as ReduxForm from 'redux-form'
-import {ADDRESS_FORM_NAME} from '../../../store/form/constants'
+
 import Button from 'progressive-web-sdk/dist/components/button'
+import {UI_NAME} from 'progressive-web-sdk/dist/analytics/data-objects/'
+
+import {ADDRESS_FORM_NAME} from '../../../store/form/constants'
+import {validatePostalCode} from '../../../utils/validation'
```
Comment (kevinxh) [inline web/app/containers/account-address/partials/address-form.jsx]: love the way you organize these `import` statements! I should do that more often! 💯

File: web/app/utils/validation.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            'validatePostalCode caught error from isPostalCode. ' +
+            'This usually occurs when isPostalCode() is passed a countryId ' +
+            `('${countryId}' in this case) that it does not support. ` +
+            '\n\n' +
+            `To fix this, update this function to validate '${countryId}' ` +
+            'postal codes manually.\n\n', e)
+        return false
+    }
+}
```
Comment (kevinxh) [inline web/app/utils/validation.js]: I sometimes see linter complains no blank line at the end of the file...

File: web/app/containers/account-address/partials/address-form.jsx (hunk 4 above)
Comment (jeremywiebe) [reply web/app/containers/account-address/partials/address-form.jsx]: It's part of my master plan: https://github.com/mobify/mobify-code-style/commit/a2c27e828e979f1fb5f5a8d8f71d58cb43e6f5f5 :)

File: web/app/utils/validation.js (hunk 5 above)
Comment (jeremywiebe) [reply web/app/utils/validation.js]: I think that's Github hiding the newline... It's there when I look at it in my local repo that's on the same commit.

**Pull request merged**